src/Scanner.h

```diff
@@ -0,0 +1,55 @@
+#ifndef SCANNER_H
+#define SCANNER_H
+
+#include <iostream>
+#include <string>
+#include <list>
+#include <map>
+#include "Token.h"
+
+
+constexpr char NULLITERAL[] = " ";
+namespace loxs {
+    class Scanner {
+        private:
+            std::string source; //Code source
+            std::list<Token> tokens; //List of tokens
+            void scanToken();
+            bool end_of_file();
+            char peek();
+            void string_literal();
+            void number_literal();
+            void identify_literal();
+            bool match(char expected);
+            char next_char();
+            long unsigned current = 0; //Points the current char of lexeme
+            long unsigned start = 0;  //Points to the first char of lexem
+            long unsigned line = 1;  //Points to the line where current is
+        public:
+            explicit Scanner(std::string code) : source(code) {}
+            std::list<Token> scanTokens();
+            void addToken(TokenType type, std::any literal);
+            void addToken(TokenType type);
+            char get_curr_char() { return source.at(current++); }
+            long unsigned get_line() { return line; }
```

Comment:
```suggestion
            long unsigned get_line() const { return line; }
```

if we need this? I don't see it's being used anywhere

Reply:
also this can be `const`

Reply:
> also this can be `const`

I believe there is no need for this to be `const` from what i read here : https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#note-56 . But correct me if I'm wrong.



> if we need this? I don't see it's being used anywhere

This will be useful in error situations, in order to inform the line where the error occurred.

Reply:
those are two different things, I can explain it in a call

Reply:
Oh yeah, adding const at the end you don't let the function to change the class members. this pointer becomes const, I am I right ?